romancal/tweakreg/astrometric_utils.py

```diff
@@ -0,0 +1,186 @@
+    through the use of the ASTROMETRIC_CATALOG_URL environment variable.
+
+    Returns
+    -------
+    ref_table : `~astropy.table.Table`
+        Astropy Table object of the catalog
+
+    """
+
+    # start by creating a composite field-of-view for all inputs
+    # This default output WCS will have the same plate-scale and orientation
+    # as the first member in the list.
+    # Fortunately, for alignment, this doesn't matter since no resampling of
+    # data will be performed.
+    if existing_wcs is not None:
+        outwcs = existing_wcs
+    else:
+        outwcs = resample_utils.make_output_wcs(input_models)
+    radius, fiducial = compute_radius(outwcs)
+
+    # perform query for this field-of-view
+    ref_dict = get_catalog(fiducial[0], fiducial[1], sr=radius, catalog=catalog)
+    colnames = ('ra', 'dec', 'mag', 'objID')
+
+    ref_table = ref_dict[colnames]
+
+    # Add catalog name as meta data
+    ref_table.meta['catalog'] = catalog
+    ref_table.meta['gaia_only'] = gaia_only
+
+    # rename coordinate columns to be consistent with tweakwcs
+    ref_table.rename_column('ra', 'RA')
+    ref_table.rename_column('dec', 'DEC')
+
+    # Append GAIA ID as a new column to the table...
+    gaia_sources = []
+    for source in ref_dict:
+        if 'GAIAsourceID' in source:
+            g = source['GAIAsourceID']
+            if gaia_only and g.strip() == '':
+                continue
+        else:
+            g = "-1"  # indicator for no source ID extracted
+        gaia_sources.append(g)
+
+    gaia_col = table.Column(data=gaia_sources, name='GaiaID', dtype='U25')
+    ref_table.add_column(gaia_col)
+
+    # sort table by magnitude, fainter to brightest
+    ref_table.sort('mag', reverse=True)
+
+    # If specified by the use through the 'num_sources' parameter,
+    # trim the returned catalog down to just the brightest 'num_sources' sources
+    # Should 'num_sources' be a negative value, it will return the faintest
+    # 'num_sources' sources.
+    if num_sources is not None:
+        indx = -1 * num_sources
+        ref_table = ref_table[:indx] if num_sources < 0 else ref_table[indx:]
+
+    # Write out table to a file, if specified
+    if output is not None:
+        ref_table.write(output, format=table_format, overwrite=True)
+
+    return ref_table
+
+
+"""
+
+Utility functions for creating an astrometric reference catalog.
+
+"""
+
+
+def compute_radius(wcs):
+    """Compute the radius from the center to the furthest edge of the WCS."""
+
+    fiducial = wcsutil.compute_fiducial([wcs], wcs.bounding_box)
+    img_center = SkyCoord(ra=fiducial[0] * u.degree, dec=fiducial[1] * u.degree)
+    wcs_foot = wcs.footprint()
+    img_corners = SkyCoord(ra=wcs_foot[:, 0] * u.degree,
+                           dec=wcs_foot[:, 1] * u.degree)
+    radius = img_center.separation(img_corners).max().value
+
+    return radius, fiducial
+
+
+def get_catalog(ra, dec, sr=0.1, catalog=DEF_CAT):
```

Comment:
Should we update this method? Currently, it is using an old API URL.
We could update the method with minimial changes by using `AstroQuery` instead of `requests`.
The new method would look like this:
```
catalog_data_csv = Catalogs.query_object(
    " ".join([str(x) for x in fiducial.tolist()]),
    radius=radius,
    catalog="GAIADR3",
    format='csv'
)
```

In the code above, `fiducial` and `radius` are the fiducial coordinates (`np.array([ra, dec])`, both in decimal degrees) and search radius (`float`, decimal degrees), respectively.

Reply:
I think we need to update this eventually, since this where in HST the proper motion & parallax adjustments are done.  But for my two cents, I would hold off on updating this until we have the step working.